detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/BitbakeGraphTransformer.java

```diff
@@ -140,4 +143,14 @@ private String removeEpochPrefix(String recipeVersion) {
 
         return Optional.ofNullable(externalId);
     }
+
+    private String chooseRecipeLayer(final String dependencyName, @Nullable String dependencyLayer, final List<String> recipeLayerNames) {
+        if (dependencyLayer == null) {
+            logger.warn("Did not parse a layer for dependency {} from task-depends.dot; falling back to layer {} (first from show-recipes output)", dependencyName, recipeLayerNames.get(0));
+            dependencyLayer = recipeLayerNames.get(0);
+        } else {
+            logger.trace("For dependency recipe {}: using layer {} parsed from task-depends.dot", dependencyName, dependencyLayer);
+        }
+        return dependencyLayer;
+    }
 }
```

Comment:
I'm half tempted to remove this fallback-to-first-in-list. I'm hoping it's never used.

Reply:
I'm perfectly fine with defaulting to the easier approach and waiting for real customer examples to come through via support. For example, assuming something doesn't happen, but throw exception if we detect it and wait for a diagnostic zip with real world example.

Reply:
I agree, we throw now as diagnostics are the only way we will get feedback. Phone home woes 😞

Reply:
@JakeMathews I'd love your opinion on this. The bitbake code does not assume it'll find the recipe version in the task-depends.dot label (GraphParseTransformer.getVersionFromNode() has returned an Optional even before I started mucking with things). Given the vast experience you have now, do you believe that caution makes sense? And should it apply to finding the layer there as well? (I realize this is largely guesswork, but I trust your guesses more than mine.)

Reply:
I think if the label is present its pretty safe to assume the layer is in the path. I always prefer to air on the side of caution especially when understanding is murky at best.
It would be great if we could phone home suspected discrepancies rather than failing or silently use a default.

i think what you have now will likely be the safest approach, but we won’t know if its right until we can collect more data.